shared/src/commonMain/kotlin/org/hyperskill/app/main/presentation/AppReducer.kt

```diff
@@ -255,15 +254,10 @@ internal class AppReducer(
         finishAction: WelcomeOnboardingFeature.Action.OnboardingFlowFinished
     ): Set<Action> =
         setOf(
-            when (val reason = finishAction.reason) {
```

Comment:
JFYI I understand that the code works as expected (like before), but in my opinion _finish reason_ previously simplified readability and understanding of the code

Reply:
@ivan-magda 
Previously, `TrackSelectionScreen` was shown only for users who passed NotificationOnboarding. Now also we should show TrackSelection for those, who have seen Paywall, but only if they are new users.
I guess, at the moment, it is easier to lean only on the profile.

Reply:
OK